### DISCUSSION TITLE
swanctl --reload-settings ,what is the meaning?

### DISCUSSION THREAD

**Discussion by ygm521** [strongswan/strongswan#1169] [General]
if i add a new connect config, and i do not want to restart charon program and also should not close tuunel before, and also let new connect config valid.  is swanctl --reload-settings  can help me?

 swanctl --reload-settings is same as follows?

```
AND  swanctl --load-authorities (-b)  (re-)load authority configuration     
AND swanctl --load-conns       (-c)  (re-)load connection configuration
AND swanctl --load-creds       (-s)  (re-)load credentials
AND swanctl --load-pools       (-a)  (re-)load pool configuration
```

also if i changed port_nat_t or port ，and want reload , how did i do?

**Comment by tobiasbrunner** [comment]
>  is swanctl --reload-settings can help me?

No, as documented that reloads settings in strongswan.conf (of which only a selected few can actually be reloaded at runtime):

    swanctl --reload-settings  (-r)  reload daemon strongswan.conf

> swanctl --reload-settings is same as follows?

No, but `swanctl --load-all` is:

    swanctl --load-all         (-q)  load credentials, authorities, pools and connections


> also if i changed port_nat_t or port ，and want reload , how did i do?

You need to restart the daemon if you do that.
